data/transactions/logic/eval.go

```diff
@@ -664,7 +665,7 @@ type EvalContext struct {
 	intc    []uint64
 	bytec   [][]byte
 	version uint64
-	Scratch scratchSpace
+	Scratch *scratchSpace
```

Comment:
Why this switch? Seems like now you'll need to allocate a Scratch instead of an EvalContext?